### DISCUSSION TITLE
Listes de serveurs vides

### DISCUSSION THREAD

**Discussion by jejevinc** [jamulussoftware/jamulus#1096] [Technical Support & Help]
Bonjour, depuis maintenant plusieurs semaines, les listes de serveurs (Genre Rock, Jazz, Folk…etc) restent vides et il semblerait que nous soyons nombreux à connaitre ce dysfonctionnement. Il reste la possibilité de se connecter par l’adresse IP, mais ça ne fonctionne pas toujours. Est-ce due au volume de connexion simultanées sur le serveur ?

**Comment by ghost** [comment]
(computer translation to english)
Hello, for several weeks now, the lists of servers (Genre Rock, Jazz, Folk… etc) remain empty and it seems that many of us are aware of this dysfunction. There is still the option to connect by the IP address, but that does not always work. Is this due to the volume of concurrent connections on the server?

**Reply by dcorson-ticino-com** [reply]
Cela semble être un problème de réseau.  Si vous laissez la fenêtre ouverte pendant un certain temps, le contenu finira par apparaître.  Mais cela semble une éternité car nous sommes habitués à une réponse immédiate. 
J'ai évoqué ce problème il y a six semaines, mais il semble que nous ne puissions pas faire grand-chose.

This seems to be a network issue.  If you leave the window open for a while the contents will eventually show up.  But it seems like an eternity as we are used to immediate response. 
I have brought up this issue 6 weeks ago, but there seems to be little that we can do.

**Comment by jejevinc** [comment]
Merci pour votre réponse. Je viens de laisser la fenêtre ouverte sur "Genre Rock" pendant presque une heure et demie et malheureusement, rien n'a changé. Merci pour votre aide. JejeVinc 
-------- Message d'origine --------De : DonC <notifications@github.com> Date : 23/02/2021  20:32  (GMT+01:00) À : jamulussoftware/jamulus <jamulus@noreply.github.com> Cc : jejevinc <cv.jerome.vincent@free.fr>, Author <author@noreply.github.com> Objet : Re: [jamulussoftware/jamulus] Listes de serveurs vides (#1096) 

Cela semble être un problème de réseau.  Si vous laissez la fenêtre ouverte pendant un certain temps, le contenu finira par apparaître.  Mais cela semble une éternité car nous sommes habitués à une réponse immédiate.

J'ai évoqué ce problème il y a six semaines, mais il semble que nous ne puissions pas faire grand-chose.

This seems to be a network issue.  If you leave the window open for a while the contents will eventually show up.  But it seems like an eternity as we are used to immediate response.

I have brought up this issue 6 weeks ago, but there seems to be little that we can do.

**Comment by hoffie** [comment]
Merci pour votre rapport! Je ne savais pas qu'il y avait des problèmes. Je ne pense pas qu'il y ait de problèmes connus de réseau / charge côté serveur.
Quelle version de Jamulus utilisez-vous?
Cela a-t-il déjà fonctionné pour vous?
Y a-t-il eu des changements dans votre configuration réseau / Internet?

La liste de serveurs utilise des paquets UDP assez volumineux qui peuvent être fragmentés. Il existe des fournisseurs d'accès Internet dans le monde qui ne gèrent pas cela correctement.
Voir ici: [Vous ne voyez pas du tout la liste des serveurs?] (Https://jamulus.io/fr/wiki/Client-Troubleshooting) 

----



Thanks for your report! I wasn't aware that there are issues. I do not think there are any known network/load problems on the server side.
Which version of Jamulus are you using?
Has it ever worked for you?
Have there been any changes in your network/Internet setup?

The server list uses rather large UDP packets which may be fragmented. There are Internet Service Providers in the world which do not handle that correctly.
See here: [Vous ne voyez pas du tout la liste des serveurs ?](https://jamulus.io/fr/wiki/Client-Troubleshooting)

**Comment by jejevinc** [comment]
Bonjour, Je suis sur la version 3.6.2. Oui cela à très bien fonctionné jusqu'à il y a quelques semaines. Pas de changement dans ma config réseau. Je suis chez free. Fibre box révolution. J'ai vu apparaître aujourd'hui la liste des serveurs Genre Rock vers 11h00 ce matin. J'ai laissé la fenêtre ouverte jusqu'à 15h00 et lorsque j'ai essayé de me connecter à un serveur tout à planté, écran bleu et redémarrage. Après redémarrage, les fenêtres sont à nouveau vides. Sniff...😢😭😤Un grand merci pour votre aide !Jérôme VINCENT06 23 90 34 26

**Comment by clivestirling** [comment]
Greetings.  I have seen a similar problem in the last few weeks: the server TheDownsmen (in genre Classical) sometimes appears in the Connection Setup window, and sometimes does not. Refreshing the list by switching Genre and back again sometimes shows it, but other times it fails to appear for any reasonable time (10-15 minutes). Many other servers do appear in the list, and can be connected to.
This happens to me and many of our group users - indeed a couple of users have never seen the server in the list (for them, connecting via IP doesn't seem to work either). Users are spread across several ISPs, I am trying to collate these to try and see if there's any pattern. One of my own ISPs is Virgin cable, and that works OK most of the time.
We are all geographiclly close to our server location (when we can see it we get a ping of 15-30)
Running 3.6.2 server and clients on Windows (and one Mac user with similar experience).
During my testing in November/December I never noticed the problem at all.
I hope this info helps solve the mystery!
Clive.

**Comment by hoffie** [comment]
Any chance that one of you can capture a tcpdump/Wireshark dump while you have the server list open? That would help to understand what the client sees (I suspect it may see partial or no data at all :( ).
You can try to filter on the port to keep the dump small and avoid any personal data ending up in it, however, it will be different per list:
- Classical would be 22524
- Rock would be 22424
----
Y a-t-il une chance que l'un de vous puisse capturer un vidage tcpdump / Wireshark pendant que la liste des serveurs est ouverte? Cela aiderait à comprendre ce que voit le client (je soupçonne qu'il peut voir des données partielles ou pas du tout :().
Vous pouvez essayer de filtrer sur le port pour garder le vidage petit et éviter que des données personnelles ne s'y retrouvent, cependant, ce sera différent selon la liste:
- Classique serait 22524
- Rock serait 22424

**Comment by jejevinc** [comment]
Désolé Christian, mais je n'ai pas ton niveau sur ces sujets. Très honnêtement, je ne comprends pas grand chose à ce que tu racontes dans ton mail et surtout, je national sais malheureusement pas faire. Désolé. Jejevinc

**Reply by hoffie** [reply]
Je suis désolé, j'ai juste sauté dans le langage technique en essayant de trouver une explication ou une solution. D'autres personnes semblent également concernées et une personne a même soumis des données que nous pouvons utiliser pour des analyses. Espérons que cela fournira des indices afin que nous puissions trouver une sorte de solution de contournement. 

----
I'm sorry, I just jumped into technical language when trying to come up with an explanation or solution. Other people seem to be affected as well and one person even submitted some data which we can use for analyses. Let's hope this provides some clue so that we can find some kind of workaround.

**Comment by dcorson-ticino-com** [comment]
Hi Christian,
As this happens to me on occasion I will try and do a dump the next time it happens.
For me at least it only happens on the servers hosted by plj, never on Volkers servers.  Maybe that is a clue.
I also see at times only partial info, for example no musician count and city.
D

**Comment by clivestirling** [comment]
Hello Christian

Thank you for your response.

I installed Wireshark and was learning how to use it then realised I may
have captured a session showing the problem.

Attached is a trace showing the following:
a. initially, about packet 226, a listing request for genre classical, the
result did not show TheDownsmen.
b. over an hour later, about packet 10724, I made a listing request for
genre rock, then
c. about packet 10741, a request for genre classical again. This time
TheDownsmen was listed, and could be connected to.
d. The capture filter was port 22524
e. I also noticed that at the time when TheDownsmen was not in the list,
any attempt to connect by IP also failed

Further info:
Jamulus client: Windows 10 running 3.6.2 (as are almost all our users),
behind a NAT router.
Jamulus server: Windows 2008R2 running 3.6.2, no NAT

Is there enough data to be useful?  Please let me know if I can help
further. I could arrange to remote into the user who has never seen
TheDownsmen in the list, for further captures.

Best wishes and many thanks
Clive

**Reply by hoffie** [reply]
> I installed Wireshark and was learning how to use it then realised I may
have captured a session showing the problem.

Cool!

> Attached is a trace showing the following:

Hrm, I fail to see a link, download or anything. Maybe I'm confused by Github's UI or was there some problem during the upload?
You may have to put such .pcap or .pcapng files into a .zip file (or rename them to .log) for Github to allow the upload.

**Comment by clivestirling** [comment]
Trace attached again here on discussion panel (last try was via email), also a newer trace capture.
[WS trace stirling 1.zip](https://github.com/jamulussoftware/jamulus/files/6054247/WS.trace.stirling.1.zip)
[WS trace stirling 3.zip](https://github.com/jamulussoftware/jamulus/files/6054248/WS.trace.stirling.3.zip)
